### PR TITLE
WIP: NEW Add a basic Behat test to set MFA to be required, and ensure "I log in as" skips MFA registration

### LINKED ISSUE #99: Introduce Behat
Introduce Behat tests to this module.

Set it up with the scaffolding and a basic test to get started.

### PR DESCRIPTION
**Pull request by robbieaverill**
We need to override the "I am logged in with X permissions" Behat context step to ensure it skips MFA registration. This is done in a class override, but all other modules with Behat tests that use this (all of them) will fail unless they are aware of MFA. This is a limitation we'll need to keep in mind when implementing CWP kitchen sink-wide Behat test runs.

Resolves #99

### CODE REVIEW

**Comment by robbieaverill** [comment]
I need to look at the build failures, but will come back to it later. I can't reproduce them locally.

File: tests/Behat/Context/LoginContext.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{
+    public function iAmLoggedInWithPermissions($permCode)
+    {
+        // Set MFA to optional, perform login logic, then skip MFA
+        $this->multiFactorAuthenticationIsOptional();
+        parent::iAmLoggedInWithPermissions($permCode);
+
+        // Wait for MFA to load
+        $this->getMainContext()->getSession()
+            ->wait(5000, 'document.getElementsByClassName("mfa-app-title").length === 1');
```
Comment (NightJar) [inline tests/Behat/Context/LoginContext.php]: Is there not another method (named `spin` or something) that is better than waiting an arbitrary `5` seconds for something to complete?
I think @ScopeyNZ has done things along this line before.

File: behat.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    mfa:
+      paths:
+        - %paths.modules.mfa%/tests/Behat/features
+      contexts:
+        - SilverStripe\Framework\Tests\Behaviour\FeatureContext
+        - SilverStripe\Framework\Tests\Behaviour\CmsFormsContext
+        - SilverStripe\Framework\Tests\Behaviour\CmsUiContext
+        - SilverStripe\BehatExtension\Context\BasicContext
+        - SilverStripe\BehatExtension\Context\EmailContext
+        - SilverStripe\MFA\Tests\Behat\Context\LoginContext
```
Comment (NightJar) [inline behat.yml]: Do behat configs work in a similar way to core config merges? Will this be sufficient to override the core login context, or does omitting it from this list achieve the same thing?

File: tests/Behat/Context/LoginContext.php (hunk 1 above)
Comment (robbieaverill) [reply tests/Behat/Context/LoginContext.php]: This is waiting _up to_ five seconds for that JavaScript condition to be true. We'd expect it to resolve much faster than that.

File: behat.yml (hunk 2 above)
Comment (robbieaverill) [reply behat.yml]: I'm not sure what you mean, but this context is overriding the core one in order to augment the behaviour of "I log in with X permission" - I've removed the core one from this list for that reason

**Comment by robbieaverill** [comment]
Will reopen this when it's working

**Pull request closed without merging**